libs/ng-utils/src/lib/directives/ng-component-outlet-augmentation.directive.ts

```diff
@@ -0,0 +1,143 @@
+}
+
+/**
+ * Augments the {@link NgComponentOutlet} directive to allow for binding of inputs and outputs.
+ *
+ * @example
+ * In your lazy loaded component define the input and outputs you wish to bind to.
+ *
+ * ```ts
+ * \@Component({
+ *  selector: 'lazy-loaded-component',
+ *  template: `...`
+ * })
+ * export class LazyLoadedComponent {
+ *  \@Input() public input1: string;
+ *
+ * \@Output() public output1 = new EventEmitter<string>();
+ * }
+ *
+ * ```
+ *
+ * In the template where you wish to use the component, use the directive to bind the inputs and outputs.
+ *
+ * ```html
+ * <ng-container *ngComponentOutlet="lazyLoadedComp$ | async; inputs: { input1: 'test' }"></ng-container>
+ * ```
+ *
+ * If you want to bind to the lazy loaded components outputs' as well, then you have to use the non-micro-syntax version of the directive, as the micro-syntax version does not support binding to outputs.
+ *
+ * ```html
+ * <ng-container
+ *  [ngComponentOutlet]="lazyLoadedComp$ | async"
+ *  [ngComponentOutletInputs]="{ input1: 'test' }"
+ *  [ngComponentOutletOutputs]="onComponentEvent($event)">
+ * </ng-container>
+ * ```
+ *
+ * Then in your component, you can handle the event like so:
+ *
+ * ```ts
+ *
+ * export class MyComponentThatUsesLazyLoadedComponent {
+ *  public onComponentEvent(event: NgComponentOutletEvent): void {
+ *    if (event.key === 'output1') {
+ *      // do something with the received event data through event.event
+ *    }
+ *  }
+ * }
+ * ```
+ */
+@Directive({
+  selector: '[ngComponentOutlet]',
+  exportAs: 'ngComponentOutlet',
+})
+export class NgComponentOutletAugmentationDirective {
+  @Input('gComponentOutlet') public component: Type<unknown> | null = null;
+
+  @Input('ngComponentOutletInjector') public injector?: Injector;
+
+  @Input('ngComponentOutletContent') public content?: Node[][];
+
+  @Input('ngComponentOutletNgModule') public ngModule?: Type<unknown>;
+
+  @Input('ngComponentOutletInputs') public inputs?: Record<string, unknown>;
+
+  @Output('ngComponentOutletOutputs') public readonly componentEvent =
+    new EventEmitter<NgComponentOutletEvent>();
+
+  private originalNgOnChanges!: (changes: SimpleChanges) => void;
+
+  public get componentRef(): ComponentRef<Record<string, unknown>> | undefined {
+    return this.ngComponentOutlet['_componentRef'];
+  }
+
+  public readonly moduleRef: NgModuleRef<unknown> | undefined;
+
+  constructor(
+    public readonly viewContainerRef: ViewContainerRef,
+    public readonly ngComponentOutlet: NgComponentOutlet
+  ) {
+    this.originalNgOnChanges = this.ngComponentOutlet.ngOnChanges.bind(
+      this.ngComponentOutlet
+    );
+    this.ngComponentOutlet.ngOnChanges = this.augmentedNgOnChanges.bind(this);
+  }
+
+  private augmentedNgOnChanges(changes: SimpleChanges): void {
+    this.originalNgOnChanges(changes);
+
+    const compChange = changes['ngComponentOutlet'];
+    if (compChange && compChange?.currentValue !== compChange?.previousValue) {
```

Comment:
It makes more sense to me:
```if (compChange && compChange.currentValue !== compChange.previousValue)```

Reply:
Did you mean a reversed order like: ...previousValue !== ...currentValue ?

Reply:
I mean the optional chaining is not necessary there and looks cleaner without it.

Reply:
then fixed

Reply:
cool :)